Plugins/DD4hep/src/ConvertDD4hepDetector.cpp

```diff
@@ -179,23 +205,12 @@ std::shared_ptr<const CylinderVolumeBuilder> volumeBuilder_dd4hep(
                                volumeDetElement.name() +
                                std::string(" has no shape!"));
       }
-      // check if it has a volume extension telling if it is a barrel or an
-      // endcap
-      ActsExtension* volumeExtension = nullptr;
-      try {
-        volumeExtension = volumeDetElement.extension<ActsExtension>();
-      } catch (std::runtime_error& e) {
-        throw std::logic_error(
-            std::string("Current DetElement: ") + volumeDetElement.name() +
-            std::string(" has no ActsExtension! At this stage it should be a "
-                        "detector volume declared as Barrel or Endcap. Please"
-                        "check your detector construction."));
-      }
 
-      if (volumeExtension->hasType("endcap", "detector")) {
-        ACTS_VERBOSE(
-            std::string("Subvolume : '") + volumeDetElement.name() +
-            std::string("' is a disc volume -> handling as an endcap"));
+      dd4hep::DetType type{volumeDetElement.typeFlag()};
+
+      if (type.is(dd4hep::DetType::ENDCAP)) {
```

Comment:
We should probably specify at some point that this is a cylindrical detector converter with barrel/EC structure.

Reply:
Absolutely. I think the documentation is pretty clear about that already though.